File: 04-Arrays/1D-Arrays/PairSum.cpp
#include <iostream>
#include <vector>
using namespace std;

//i/p
// 7       
// 1 3 5 7 2 4 6
// 9

//o/p
// 3 6
// 5 4
// 7 2

void PairSum(vector<int> arr,int sum){

    for(int i = 1 ; i <= arr.size(); i++)
    {
        for(int j = i + 1  ; j <= arr.size() ;j++ ){
            if(arr[i] + arr[j] == sum){
                cout<<arr[i] << " "<< arr[j]<<endl;
            }
        }
    }
    
}

int main() {

    int n1;
    cin>>n1;
    vector<int> arr(n1);
    
    for(int i = 0 ; i< n1 ; i++){
        cin>>arr[i];
    }

    int sum;
    cin>>sum;


   PairSum(arr,sum);
    
 

    return 0;
}

File: 05-Array_List-InPython/1D Array in python/PairSum.py
# //i/p
# // 7       
# // 1 3 5 7 2 4 6
# // 9

# //o/p
# // 3 6
# // 5 4
# // 7 2

def PairSum(arr, sum_val):
    for i in range(len(arr)):
        for j in range(i + 1, len(arr)):
            if arr[i] + arr[j] == sum_val:
                print(arr[i], arr[j])

def main():
    n = int(input())
    # arr = list(map(int, input().split()))
    arr = list(map(int, input().split()))
    sum_val = int(input())

    PairSum(arr, sum_val)

if __name__ == "__main__":
    main()
